### PR TITLE
[bitnami/phpbb] MAJOR:  Move to non-root

### PR DESCRIPTION
**Pull request by javsalgar**
**Description of the change**

We are working on releasing the phpBB Docker image with a non-root approach. As part of that, we need to update the chart to use the new port numbers used by the container image and related settings.

We're also taking advantage of this PR being a major change by adding all standarizations (i.e. customizable command, extra volumes, extra envs, K8s labels standarization, etc.).

**Benefits**

Support for environments such as OpenShift, where a non-root UID is required to run a container.

**Possible drawbacks**

Breaking change with previous chart versions, as the default HTTP/HTTPS ports are changed to 8080/8443 (previously 80/443). The labels may also cause this to be a breaking change.

**Applicable issues**

  - fixes #

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/phpbb/templates/NOTES.txt [hunk 1]
```diff
@@ -1,54 +1,84 @@
...
+  {{- else }}
+    {{- range .Values.ingress.hosts }}
+    - http://{{ . }}
+    {{- end }}
+  {{- end }}
+{{- else if contains "NodePort" .Values.service.type }}
+  export NODE_PORT=$(kubectl get --namespace {{ .Release.Namespace }} -o jsonpath="{.spec.ports[0].nodePort}" services {{ template "phpbb.fullname" . }})
   export NODE_IP=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath="{.items[0].status.addresses[0].address}")
-  echo "phpBB URL: http://$NODE_IP:$NODE_PORT/"
+  echo "phpBB! URL: http://$NODE_IP:$NODE_PORT/"
```
Comment (marcosbc) [inline bitnami/phpbb/templates/NOTES.txt:23]: I see many occurrences of "phpBB!" as if the product name includes the "!", it does not look to be the case so could you remove it?

File: bitnami/phpbb/templates/deployment.yaml [hunk 2]
```diff
@@ -1,166 +1,240 @@
...
+            timeoutSeconds: {{ .Values.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.livenessProbe.failureThreshold }}
+          {{- else if .Values.customLivenessProbe }}
+          livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.customLivenessProbe "context" $) | nindent 12 }}
+          {{- end }}
+          {{- if .Values.readinessProbe.enabled }}
+          readinessProbe:
+            httpGet:
+              path: /index.php
```
Comment (marcosbc) [inline bitnami/phpbb/templates/deployment.yaml:186]: I think that loading the index.php could prove to be a huge load, note that it is the entire forum index and many SQL queries are performed. Could you find a different path?

File: bitnami/phpbb/templates/deployment.yaml [hunk 3]
```diff
@@ -1,166 +1,242 @@
...
+          {{- end }}
+          ports:
+            - name: http
+              containerPort: {{ .Values.containerPorts.http }}
+            - name: https
+              containerPort: {{ .Values.containerPorts.https }}
+          {{- if .Values.livenessProbe.enabled }}
+          livenessProbe:
+            httpGet:
+              path: /app.php/help/faq
```
Comment (marcosbc) [inline bitnami/phpbb/templates/deployment.yaml:175]: Could you confirm that the FAQ will try and connect to the database?

Comment (javsalgar) [reply bitnami/phpbb/templates/deployment.yaml:175]: It fails if there's no database

**Review by marcosbc — APPROVED** [review]
Awesome, thanks! LGTM

**Pull request merged**